src/autowiring/AutoConfigListing.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright (C) 2012-2015 Leap Motion, Inc. All rights reserved.
+#include "stdafx.h"
+#include "AutoConfigBase.h"
+#include "AutoConfigListing.h"
+#include "AutoConfigParser.hpp"
+#include "demangle.h"
+#include <sstream>
+
+using namespace autowiring;
+
+// Create map of all config values in the registry
+static std::unordered_map<std::string, const ConfigRegistryEntry*> FillRegistry(void) {
+  std::unordered_map<std::string, const ConfigRegistryEntry*> registry;
+  
+  for (auto config = g_pFirstConfigEntry; config; config = config->pFlink) {
+    registry[config->m_key] = config;
+  }
+  
+  return registry;
+}
+
+// Create map of all validators specified
+static std::unordered_map<std::string, AutoConfigListing::t_validator> FillValidators(void) {
+  std::unordered_map<std::string, AutoConfigListing::t_validator> validator;
+  
+  for (auto config = g_pFirstConfigEntry; config; config = config->pFlink) {
+    if (config->m_hasValidator) {
+      validator[config->m_key] = config->validator();
+    }
+  }
+  
+  return validator;
+}
+
+const std::unordered_map<std::string, const ConfigRegistryEntry*> AutoConfigListing::s_registry = FillRegistry();
```

Comment:
Note to switch this to a lambda function, especially if you don't intend this to be used elsewhere.